functions/Remove-DbaBackup.ps1

```diff
@@ -99,143 +99,158 @@ It will also remove any backup folders that no longer contain backup files.
+			The last character of the string will indicate units (validated)
+			Valid units are: (h = hours, d = days, w = weeks, m = months)
+
+			The preceeding characters are the value and must be an integer (validated)
+
+			Examples:
+				'48h' = 48 hours
+				'7d' = 7 days
+				'4w' = 4 weeks
+				'1m' = 1 month
+			#>
+
+			[int]$Length = ($UserFriendlyRetention).Length
+			$Value = ($UserFriendlyRetention).Substring(0,$Length-1)
+			$Units = ($UserFriendlyRetention).Substring($Length-1,1)
+
+			# Validate that $Units is an accepted unit of measure
+			if ( $Units -notin @('h','d','w','m') ) {
+				throw "RetentionPeriod '$UserFriendlyRetention' units invalid! See Get-Help for correct formatting and examples."
+			}
+
+			# Validate that $Value is an INT
+			if ( ![int]::TryParse($Value,[ref]"") ) {
+				throw "RetentionPeriod '$UserFriendlyRetention' format invalid! See Get-Help for correct formatting and examples."
+			}
+
+			switch ($Units)
+			{
+				'h' { $UnitString = 'Hours'; [datetime]$ReturnDatetime = (Get-Date).AddHours(-$Value)  }
+				'd' { $UnitString = 'Days';  [datetime]$ReturnDatetime = (Get-Date).AddDays(-$Value)   }
+				'w' { $UnitString = 'Weeks'; [datetime]$ReturnDatetime = (Get-Date).AddDays(-$Value*7) }
+				'm' { $UnitString = 'Months';[datetime]$ReturnDatetime = (Get-Date).AddMonths(-$Value) }
+			}
+			$ReturnDatetime
+		}
+
+		# Validations
+		# Ensure BackupFileExtension does not begin with a .
+		if ($BackupFileExtension -match "^[.]") {
+			Write-Message -Message "Parameter -BackupFileExtension begins with a period '$BackupFileExtension'. A period is automatically prepended to -BackupFileExtension and need not be passed in." -Warning
+		}
+
+	}
+	PROCESS
+	{
+		# Process stuff
+		Write-Message -Message "Started" -Level 3 -Silent $Silent
+		Write-Message -Message "Removing backups from $Path" -Level 3 -Silent $Silent
+		# Convert Retention Value to an actual DateTime
+		try {
+			$RetentionDate = Convert-UserFriendlyRetentionToDatetime -UserFriendlyRetention $RetentionPeriod
+			Write-Message -Message "Backup Retention Date set to $RetentionDate" -Level 5 -Silent $Silent
+		} catch {
+			throw $_
+		}
+
+		# Filter out unarchived files if -CheckArchiveBit parameter is used
+		if ($CheckArchiveBit.IsPresent) {
+			Write-Message -Message "Removing only archived files" -Level 5 -Silent $Silent
+			Filter DbaArchiveBitFilter {
+				If ($_.Attributes -notmatch "Archive") {
+					$_
+				}
+			}
+		} else {
+			Filter DbaArchiveBitFilter {
+				$_
+			}
+		}
+		# Enumeration may take a while. Without resorting to "esoteric" file listing facilities
+		# and given we need to fetch at least the LastWriteTime, let's just use "streaming" processing
+		# here to avoid issues like described in #970
+		Get-ChildItem $Path -Filter "*.$BackupFileExtension" -File -Recurse -ErrorAction SilentlyContinue -ErrorVariable EnumErrors | `
+			Where-Object LastWriteTime -lt $RetentionDate | DbaArchiveBitFilter | Foreach-Object `
```

Comment:
Avoid using backticks to break lines when things get long.
Either allow code to go wide (totally fine with me) or use the ability to automatically move to the next line with a pipe symbol:
```PowerShell
Get-ChildItem $Path -Filter "*.$BackupFileExtension" -File -Recurse -ErrorAction SilentlyContinue -ErrorVariable EnumErrors |
    Where-Object LastWriteTime -lt $RetentionDate |
        DbaArchiveBitFilter |
            Foreach-Object {
```